### ISSUE TITLE
Assembly loading fails for Microsoft.CodeAnalysis.CSharp.resources with Culture=neutral on non-English systems

### ISSUE THREAD

**Issue by cjbor** [dotnet/roslyn#81168]
## Problem

When using Roslyn 4.14.0 at runtime on systems with non-English culture settings, 
.NET attempts to load:

`Microsoft.CodeAnalysis.CSharp.resources, Culture=neutral, PublicKeyToken=null`

This assembly does not exist and should not be requested. The neutral (English) 
resources should be embedded in the main assembly.

## Reproduction

```csharp
// On a system with German/French/etc Windows:
using Microsoft.CodeAnalysis.CSharp;
using Microsoft.CodeAnalysis.CSharp.Scripting;
// This triggers the error:
var script = CSharpScript.Create("1 + 1");
``` 

**Error**

Could not load file or assembly 'Microsoft.CodeAnalysis.CSharp.resources, Culture=neutral, PublicKeyToken=null'. The system cannot find the specified file.

## Root Cause

The `NeutralResourcesLanguage` assembly attribute is either:
1. Missing
2. Incorrectly configured without `UltimateResourceFallbackLocation.MainAssembly`

This causes .NET's ResourceManager to search for a satellite assembly with 
neutral culture instead of using the resources embedded in the main assembly.

### Option 1: Add/Fix Assembly Attribute (Recommended)

In `src/Compilers/CSharp/Portable/CSharpResources.Designer.cs` or assembly-level file:

```csharp
[assembly: NeutralResourcesLanguage("en", UltimateResourceFallbackLocation.MainAssembly)]
```

Apply the same to:
- `Microsoft.CodeAnalysis`
- `Microsoft.CodeAnalysis.CSharp.Workspaces`
- `Microsoft.CodeAnalysis.CSharp.Scripting`
- `Microsoft.CodeAnalysis.VisualBasic` (and related)
- `Microsoft.CodeAnalysis.Workspaces`
- `Microsoft.CodeAnalysis.Scripting`

### Option 2: MSBuild Configuration

Ensure `.csproj` files specify:

```csproj
<SatelliteResourceLanguages>en de;es;fr;it;ja;ko;pl;pt-BR;ru;tr;zh-Hans;zh-Hant</SatelliteResourceLanguages>
```

(Note: `en` is deliberately excluded to keep it in the main assembly)

## Workaround
Users currently need to add this to their application startup:

```csharp
AppDomain.CurrentDomain.AssemblyResolve += (sender, args) =>
{
    var name = new AssemblyName(args.Name);
    if (name.Name?.Contains("Microsoft.CodeAnalysis") == true && name.Name.EndsWith(".resources"))
    {
        return Assembly.Load(name.Name.Replace(".resources", ""));
    }
    return null;
};
```

Or add to their `.csproj`:
```csproj
<SatelliteResourceLanguages>en</SatelliteResourceLanguages>
```

## Impact

This affects any application using Roslyn at runtime on non-English systems:
- Scripting scenarios
- Runtime code generation frameworks (e.g., WolverineFx, NServiceBus)
- Code analyzers
- IDE extensions

## Environment

- Roslyn: 4.14.0 (and likely earlier versions)
- .NET: 8.0, 9.0
- OS: Windows with non-English culture settings

## References

- [.NET Satellite Assembly Loading](https://learn.microsoft.com/en-us/dotnet/core/dependency-loading/loading-resources)
- [NeutralResourcesLanguage Attribute](https://learn.microsoft.com/en-us/dotnet/api/system.resources.neutralresourceslanguageattribute)

**Comment by jjonescz** [comment]
Thanks for taking the time to report this issue.

I cannot repro from the provided information. Specifically, I've set my Windows language to Czech and ran this file-based app:

```cs
#:package Microsoft.CodeAnalysis.CSharp.Scripting@4.14.0
using Microsoft.CodeAnalysis.CSharp.Scripting;
var script = CSharpScript.Create("1 + 1");
var result = await script.RunAsync();
Console.WriteLine(result.ReturnValue);
```

I got no errors.

Can you provide a complete minimal repro?

**Comment by Sebazzz** [comment]
I'm also running into this issue, when referencing 4.14.0 nothing from that assembly (things like `DiagnosticDescriptor`) can be resolved.

**Comment by jjonescz** [comment]
Can you provide a minimal reproducible example so we can investigate further? A [complog](https://github.com/jaredpar/complog?tab=readme-ov-file#creating-compiler-logs) might be ideal.

**Comment by cjbor** [comment]
I tried building a simple source generator only importing Microsoft.CodeAnalysis.CSharp and ran into the same exception.
The complog from my machine is attached.

[SourceGen.zip](https://github.com/user-attachments/files/24736314/SourceGen.zip)

**Comment by jjonescz** [comment]
Thanks, @cjbor. Can you provide also the binlog that was used to produce that complog? Which exact culture do you have set?

I have also noticed the original issue have had weird formatting (it looks like you copy pasted it from somewhere, but keep in mind that GitHub renders issues as Markdown and some constructs like angle brackets get lost in rendering if they are outside a code block). I have tried to fix it up where I could but you may want to re-check.

**Comment by cjbor** [comment]
I reduced the exaple even more with almost no custom code and switched to netstandard2.0 as this was sugested in multiple places.
The Exception remains the same. complog + binlog included. IDE is Rider 2025.3.1.

Using Get-Culture in Powershell

LCID      |       Name        |     DisplayName
1031     |       de-DE        |    Deutsch (Deutschland)

[SourceGen.zip](https://github.com/user-attachments/files/24738062/SourceGen.zip)

**Comment by jjonescz** [comment]
I don't see any errors in the binlog. I thought there was supposed to be an exception during build, or did I misunderstand?

**Comment by cjbor** [comment]
Its a runtime excetion as soon as I start debugging.

**Comment by jjonescz** [comment]
Does it happen only in Rider or in VS or VSCode too?

**Comment by cjbor** [comment]
<img width="915" height="630" alt="Image" src="https://github.com/user-attachments/assets/faedef7d-797e-4491-be27-43924b7d189c" />

As of right now im not able to find a method to start the source gen with a debugger attached.

**Comment by jjonescz** [comment]
> As of right now im not able to find a method to start the source gen with a debugger attached.

As a temporary workaround, you can try calling `Debugger.Launch()` from the source generator.

**Comment by jjonescz** [comment]
I'm getting that error you posted when the targetProject specified in the launchSettings.json doesn't exist and isn't part of the open solution. Otherwise I cannot repro this. Please verify your config is correct.
